src/widgets/Updater.h

```diff
@@ -0,0 +1,144 @@
+﻿/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  Updater.h
+
+  Dominic Mazzoni
+
+**********************************************************************/
+
+#ifndef __AUDACITY_UPDATER__
+#define __AUDACITY_UPDATER__
+
+#include "ViewInfo.h" // for children
+#include "Envelope.h"
+#include "NumberScale.h" // member variable
+#include <wx/font.h>
+
+class wxDC;
+class wxColor;
+
+enum RulerFormat {
+   IntFormat,
+   RealFormat,
+   RealLogFormat,
+   TimeFormat,
+   LinearDBFormat,
+};
+
+struct RulerStruct {
+   struct Fonts {
+      wxFont major, minor, minorMinor;
+      int lead;
+   };
+
+   double mMin{ 0.0 };
+   double mHiddenMin{ 0.0 };
+   double mMax{ 100.0 };
+   double mHiddenMax{ 100.0 };
+
+   int mOrientation{ wxHORIZONTAL };
+   int mSpacing{ 6 };
+   RulerFormat mFormat{ RealFormat };
+   bool mFlip{ false };
+   bool mLabelEdges{ false };
+
+   int mLeft{ -1 };
+   int mTop{ -1 };
+   int mRight{ -1 };
+   int mBottom{ -1 };
+   int mLength{ 0 };
+
+   double mDbMirrorValue{ 0.0 };
+
+
+   mutable std::unique_ptr<Fonts> mpFonts;
+   TranslatableString mUnits;
+
+   int mLeftOffset;
+
+   NumberScale mNumberScale;
```

Comment:
Reviewing the commit "Remove circular dependencies..." which moves fields into the new type `RulerStruct` -- I see that the member `mLeftOffset` is never initialized.

A later commit removes the constructor in favor of in-class initializers, which is all right, but still does not correct this.

The remaining fields without in-class initializers are all right to remain so, because they are not primitive types.  So their default constructors are implicitly invoked.  You could write `{}` to initialize each but it isn't needed and makes no difference in compiled code.  I omit such initializers myself in such cases.